hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java

```diff
@@ -0,0 +1,418 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.om;
+
+import com.google.protobuf.RpcController;
+import com.google.protobuf.ServiceException;
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.hdds.utils.db.cache.CacheKey;
+import org.apache.hadoop.hdds.utils.db.cache.CacheValue;
+import org.apache.hadoop.ozone.OFSPath;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyArgs;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OzoneFileStatus;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.util.Progressable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.apache.hadoop.ozone.OzoneConsts.OZONE_URI_DELIMITER;
+import static org.apache.hadoop.ozone.om.helpers.OzoneFSUtils.addTrailingSlashIfNeeded;
+import static org.apache.hadoop.ozone.om.helpers.OzoneFSUtils.pathToKey;
+import static org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.Type.DeleteKeys;
+
+/**
+ * FileSystem to be used by the Trash Emptier.
+ * Only the apis used by the trash emptier are implemented.
+ */
+public class TrashOzoneFileSystem extends FileSystem {
+
+  private static final RpcController NULL_RPC_CONTROLLER = null;
+
+  private static final int OZONE_FS_ITERATE_BATCH_SIZE = 100;
+
+  private OzoneManager ozoneManager;
+
+  private String userName;
+
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(TrashOzoneFileSystem.class);
+
+  public TrashOzoneFileSystem(OzoneManager ozoneManager) throws IOException {
+    this.ozoneManager = ozoneManager;
+    this.userName =
+          UserGroupInformation.getCurrentUser().getShortUserName();
+  }
+
+  @Override
+  public URI getUri() {
+    return null;
+  }
+
+  @Override
+  public FSDataInputStream open(Path path, int i) throws IOException {
+    return null;
+  }
+
+  @Override
+  public FSDataOutputStream create(Path path,
+      FsPermission fsPermission,
+      boolean b, int i, short i1,
+      long l, Progressable progressable) throws IOException {
+    return null;
+  }
+
+  @Override
+  public FSDataOutputStream append(Path path, int i, Progressable progressable)
+      throws IOException {
```

Comment:
In order to remove the null/empty functions, I feel we can stop import from Filesystem.java